hphp/runtime/vm/jit/vasm-fold-imms.cpp

```diff
@@ -413,6 +420,17 @@ struct ImmFolder {
   void fold(cmpl& in, Vinstr& out) { return fold_cmp<cmpli>(in, out); }
   void fold(cmpq& in, Vinstr& out) { return fold_cmp<cmpqi>(in, out); }
 
+  void fold(andq& in, Vinstr& out) {
+    int val;
+    if (logical_imm(in.s0, val)) { out = andqi{val, in.s1, in.d, in.sf}; }
+    else if (logical_imm(in.s1, val)) { out = andqi{val, in.s0, in.d, in.sf}; }
+    else {
+      uint64_t bm;
+      if (logical_BitMask(in.s0, bm)) { out = andzi{bm, in.s1, in.d, in.sf}; }
+      else if (logical_BitMask(in.s1, bm)) { out = andzi{bm, in.s0, in.d, in.sf}; }
```

Comment:
This line is > 80 chars long.

I think it's probably better to just change everything in this function to this sort of thing:

```
if (...) {
  // stuff
} else if (...) {
  // stuff
} else {
  // stuff
}
```

etc.